File: ios/MenCare/Core/DesignSystem/Component/Buttons.swift
import SwiftUI

struct AccentButton: View {
    let text: String
    let action: () -> Void
    var enabled: Bool = true
    var loading: Bool = false
    var height: CGFloat = 52
    var corner: CGFloat = 14
    var leadingSystemImage: String?
    /// L'accento ha due facce: oliva sul chiaro, oro sulle bande scure. Il
    /// bottone è lo stesso, cambia solo su cosa è appoggiato.
    var container: Color = .oliveWood
    var contentColor: Color = .bone

    var body: some View {
        Button(action: action) {
            HStack(spacing: 9) {
                if loading {
                    ProgressView().tint(contentColor)
                } else {
                    if let leadingSystemImage {
                        Image(systemName: leadingSystemImage).font(.system(size: 15))
                    }
                    Text(text).font(Typo.titleMedium).kerning(0.8)
                }
            }
            .frame(maxWidth: .infinity)
            .frame(height: height)
        }
        .buttonStyle(.plain)
        .foregroundStyle(contentColor)
        .background(RoundedRectangle(cornerRadius: corner).fill(container))
        .opacity(enabled && !loading ? 1 : 0.5)
        .disabled(!enabled || loading)
    }
}

/// Social sign-in button: brand glyph on the Stone surface.
struct SocialButton: View {
    let text: String
    let icon: Image
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack(spacing: 9) {
                icon.resizable().scaledToFit().frame(width: 16, height: 16)
                Text(text).font(Typo.titleSmall)
            }
            .frame(maxWidth: .infinity)
            .frame(height: 52)
        }
        .buttonStyle(.plain)
        .foregroundStyle(Color.ink)
        .background(RoundedRectangle(cornerRadius: 16).fill(Color.stone))
    }
}

File: android/core/designsystem/src/main/kotlin/com/devora/mencare/core/designsystem/component/Buttons.kt
package com.devora.mencare.core.designsystem.component

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.Image
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Shape
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.em
import com.devora.mencare.core.designsystem.theme.Bone
import com.devora.mencare.core.designsystem.theme.Ink
import com.devora.mencare.core.designsystem.theme.OliveWood
import com.devora.mencare.core.designsystem.theme.Radii
import com.devora.mencare.core.designsystem.theme.Stone

@Composable
fun AccentButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
    loading: Boolean = false,
    height: Dp = 52.dp,
    shape: Shape = MaterialTheme.shapes.medium,
    leadingIcon: ImageVector? = null,
    // L'accento ha due facce: oliva sul chiaro, oro sulle bande scure. Il
    // bottone è lo stesso, cambia solo su cosa è appoggiato.
    container: Color = OliveWood,
    contentColor: Color = Bone,
) {
    Button(
        onClick = onClick,
        modifier = modifier.fillMaxWidth().height(height),
        enabled = enabled && !loading,
        shape = shape,
        colors = ButtonDefaults.buttonColors(containerColor = container, contentColor = contentColor),
    ) {
        if (loading) {
            CircularProgressIndicator(
                modifier = Modifier.height(20.dp),
                color = MaterialTheme.colorScheme.onPrimary,
                strokeWidth = 2.dp,
            )
        } else {
            if (leadingIcon != null) {
                Icon(leadingIcon, contentDescription = null, modifier = Modifier.size(17.dp))
                Spacer(Modifier.width(9.dp))
            }
            Text(text, style = MaterialTheme.typography.titleMedium.copy(letterSpacing = 0.05.em))
        }
    }
}

/** Social sign-in button: brand glyph on the Stone surface. */
@Composable
fun SocialButton(
    text: String,
    iconRes: Int,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Button(
        onClick = onClick,
        modifier = modifier.fillMaxWidth().height(52.dp),
        shape = RoundedCornerShape(16.dp),
        colors = ButtonDefaults.buttonColors(containerColor = Stone, contentColor = Ink),
        elevation = null,
    ) {
        Image(
            painter = painterResource(iconRes),
            contentDescription = null,
            modifier = Modifier.size(16.dp),
        )
        Spacer(Modifier.width(9.dp))
        Text(text, style = MaterialTheme.typography.titleSmall)
    }
}

@Composable
fun SecondaryButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
    onDark: Boolean = false,
    height: Dp = 52.dp,
    leadingIcon: ImageVector? = null,
) {
    val content = if (onDark) Bone else Ink
    OutlinedButton(
        onClick = onClick,
        modifier = modifier.fillMaxWidth().height(height),
        enabled = enabled,
        shape = Radii.Md,
        colors = ButtonDefaults.outlinedButtonColors(contentColor = content),
        border = BorderStroke(1.5.dp, if (onDark) Bone.copy(alpha = 0.4f) else content),
    ) {
        if (leadingIcon != null) {
            Icon(leadingIcon, contentDescription = null, modifier = Modifier.size(17.dp))
            Spacer(Modifier.width(9.dp))
        }
        Text(text, style = MaterialTheme.typography.titleMedium.copy(letterSpacing = 0.05.em))
    }
}

@Composable
fun LinkButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    contentPadding: PaddingValues = ButtonDefaults.TextButtonContentPadding,
) {
    TextButton(onClick = onClick, modifier = modifier, contentPadding = contentPadding) {
        Text(
            text,
            style = MaterialTheme.typography.labelLarge,
            color = OliveWood,
        )
    }
}
